consumer-services/src/main/scala/za/co/absa/spline/consumer/service/repo/DataSourceRepositoryImpl.scala

```diff
@@ -96,16 +96,17 @@ class DataSourceRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) extends Dat
         |    RETURN resItem
         |""".stripMargin,
       Map(
-        "asAtTime" -> (asAtTime: java.lang.Long),
-        "timestampStart" -> (writeTimestampStart: java.lang.Long),
-        "timestampEnd" -> (writeTimestampEnd: java.lang.Long),
-        "pageOffset" -> (pageRequest.page - 1: Integer),
-        "pageSize" -> (pageRequest.size: Integer),
+        "asAtTime" -> Long.box(asAtTime),
+        "timestampStart" -> Long.box(writeTimestampStart),
+        "timestampEnd" -> Long.box(writeTimestampEnd),
+        "pageOffset" -> Int.box(pageRequest.page - 1),
+        "pageSize" -> Int.box(pageRequest.size),
```

Comment:
Just curious: Why is the boxing needed here?

Reply:
Because the `db.queryAs()` method expects Map[String, AnyRef]`, but when you put a value of a primitive type to the map, the type is inferred as `Map[String, Any]` instead. Previously I've used type ascription to induce an auto boxing, but now it occurs to me that explicitly calling `box()' or `unbox()` method on respective type companion object is cleaner and more readable. Especially in other places, e.g. when doing `Option.map()` for example:
```scala
Option(42).map(x => x : java.lang.Integer)
// versus
Option(42).map(Int.box)
```